### PR TITLE
[SPARK-40675][DOCS] Supplement missing spark k8s configurations in `running-on-kubernetes.md`

### PR DESCRIPTION
**Pull request by dcoliversun**
### What changes were proposed in this pull request?

Supplement missing spark k8s configuration in `running-on-kubernetes.md`.

### Why are the changes needed?

Help users to confirm configurations about k8s through doc instead of doc.

### Does this PR introduce _any_ user-facing change?

Yes, more configurations in doc.

### How was this patch tested?

Pass the GA.

### CODE REVIEW

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -1274,6 +1274,14 @@ See the [configuration page](configuration.html) for information on Spark config
   </td>
   <td>3.0.0</td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.trust.certificates</code></td>
```
Comment (dcoliversun) [inline docs/running-on-kubernetes.md:1278]: https://github.com/apache/spark/blob/6865309604a9986d902a6ff145f0855ee3fb7f8f/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala#L651-L656

File: docs/running-on-kubernetes.md [hunk 2]
```diff
@@ -1395,6 +1403,22 @@ See the [configuration page](configuration.html) for information on Spark config
   </td>
   <td>3.2.0</td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.driver.service.ipFamilyPolicy</code></td>
```
Comment (dcoliversun) [inline docs/running-on-kubernetes.md:1407]: https://github.com/apache/spark/blob/6865309604a9986d902a6ff145f0855ee3fb7f8f/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala#L66-L72

File: docs/running-on-kubernetes.md [hunk 3]
```diff
@@ -1395,6 +1403,22 @@ See the [configuration page](configuration.html) for information on Spark config
...
+  <td>SingleStack</td>
+  <td>
+    K8s IP Family Policy for Driver Service.
+  </td>
+  <td>3.4.0</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.driver.service.ipFamilies</code></td>
```
Comment (dcoliversun) [inline docs/running-on-kubernetes.md:1415]: https://github.com/apache/spark/blob/6865309604a9986d902a6ff145f0855ee3fb7f8f/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala#L74-L80

File: docs/running-on-kubernetes.md [hunk 4]
```diff
@@ -1501,6 +1525,14 @@ See the [configuration page](configuration.html) for information on Spark config
   </td>
   <td>3.3.0</td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.executor.eventProcessingInterval</code></td>
```
Comment (dcoliversun) [inline docs/running-on-kubernetes.md:1529]: https://github.com/apache/spark/blob/6865309604a9986d902a6ff145f0855ee3fb7f8f/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala#L523-L531

File: docs/running-on-kubernetes.md [hunk 5]
```diff
@@ -1395,6 +1403,22 @@ See the [configuration page](configuration.html) for information on Spark config
   </td>
   <td>3.2.0</td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.driver.service.ipFamilyPolicy</code></td>
+  <td>SingleStack</td>
```
Comment (Yikun) [inline docs/running-on-kubernetes.md:1408]: nits:
```suggestion
  <td><code>SingleStack</code></td>
```

**Comment by dcoliversun** [comment]
GA works normally with 85ddc806bf92e8492d41aa051b8f836da4c7757d . Thanks for @Yikun work

File: docs/running-on-kubernetes.md [hunk 6]
```diff
@@ -1395,6 +1403,22 @@ See the [configuration page](configuration.html) for information on Spark config
...
+  <td>
+    K8s IP Family Policy for Driver Service.
+  </td>
+  <td>3.4.0</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.driver.service.ipFamilies</code></td>
+  <td>IPv4</td>
```
Comment (Yikun) [inline docs/running-on-kubernetes.md:1416]: nits:
```suggestion
  <td><code>IPv4</code></td>
```

File: docs/running-on-kubernetes.md (hunk 5 above)
Comment (dcoliversun) [reply docs/running-on-kubernetes.md]: Thanks, done

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @dcoliversun and @Yikun .
Merged to master for Apache Spark 3.4.0.

**Comment by dongjoon-hyun** [comment]
- To merge this, I removed `(part 2)` from the title.
- Since there is Apache Spark 3.4 configuration, I merged to `master` branch only.

**Comment by dcoliversun** [comment]
Thanks all @dongjoon-hyun and @Yikun

**Pull request closed without merging**